File: python/lesson 8/dfs_graph.py
class DfsGraph:
    def traverse(self, graph: Graph):
        visited = [False] * (graph.size())
        for v in range(0, graph.size() - 1):
            self.dfs(graph, v, visited)
            
    def dfs(self, graph: Graph, v: int, visited: List[bool]):
        if visited[v]:
            return
        
        visited[v] = True
        print(v)
        
        for w in graph.get(v):
            self.dfs(graph, w, visited)

File: java/lesson 8/DfsGraph.java
class DfsGraph {
    public void dfs(DirectedGraph graph) {
        boolean[] visited = new boolean[graph.size()];
        for (int v = 0; v < graph.size(); v++) {
            dfs(graph, v, visited);
        }
    }
    
    private void dfs(DirectedGraph graph, int v, boolean[] visited) {
        if (visited[v]) return; 
        
        visited[v] = true;
        System.out.println(v);
        
        for (int w: graph.get(v)) {
           dfs(graph, w, visited);
        }
    }
}
